taf/testlib/virtual_env.py

```diff
@@ -310,17 +376,19 @@ def ensure_public_access(self, try_reuse=False, networks_client=None, routers_cl
 
         _net_cfg.public_network_id = public_network['id']
 
-        router_kwargs = {
-            'routers_client': routers_client,
-            'network_id': _net_cfg.public_network_id,
-            'tenant_id': tenant_id,
-            'enable_snat': True
-        }
-        public_router = self.create_router(**router_kwargs)
-        assert public_router
-        _net_cfg.public_router_id = public_router['id']
+        if create_external_router:
+            router_kwargs = {
+                'routers_client': routers_client,
+                'network_id': _net_cfg.public_network_id,
+                'tenant_id': tenant_id,
+                'enable_snat': True
+            }
+            public_router = self.create_router(**router_kwargs)
+            assert public_router
```

Comment:
Yes, that's right. This assert is meant only to catch the situation when the cluster creation fails and the tempest returns None. We could test a lot of things to make sure that we got a correct dictionary. But if it woudn't have e.g. the id key, something would be very wrong and then I think the KeyError exception or similar is appropriate.